account.go

```diff
@@ -92,30 +116,34 @@ func (account *Account) HashMultiCall(fee *types.Felt, nonce *big.Int, calls []t
 func (account *Account) EstimateFee(ctx context.Context, calls []types.Transaction) (*types.FeeEstimate, error) {
```

Comment:
I think we want to pass the `ExecutionDetails` here too, otherwise the estimate fee payload is different from the subsequent call that uses it